File: Lenguajes Orientados a Objetos/Parcial 2 Actividad 4/Python/EmpleadoAsalariado.py
from Empleado import Empleado


class EmpleadoAsalariado(Empleado):
    def __init__(self, nombre, apellido, nss, salario):
        super().__init__(nombre, apellido, nss)
        self.salario_semanal = salario

    @property
    def salario_semanal(self):
        return self._salario_semanal

    @salario_semanal.setter
    def salario_semanal(self, salario):
        if salario < 0.0:
            self._salario_semanal = 0.0
        else:
            self._salario_semanal = salario

    def ingresos(self):
        return self.salario_semanal

    def __str__(self):
        return f"empleado asalariado: {super().__str__()}\nsalario semanal: ${self.salario_semanal:,.2f}"

File: Lenguajes Orientados a Objetos/Parcial 2 Actividad 4/C#/EmpleadoAsalariado.cs
namespace SistemaNomina
{
    public class EmpleadoAsalariado : Empleado
    {
        private decimal _salarioSemanal;

        public decimal SalarioSemanal
        {
            get { return _salarioSemanal; }
            set { _salarioSemanal = value < 0 ? 0 : value; }
        }

        public EmpleadoAsalariado(string nombre, string apellido, string nss, decimal salario)
            : base(nombre, apellido, nss)
        {
            SalarioSemanal = salario;
        }

        public override decimal Ingresos() => SalarioSemanal;

        public override string ToString()
        {
            return $"empleado asalariado: {base.ToString()}\nsalario semanal: {SalarioSemanal:C}";
        }
    }
}



